server/src/main/java/com/cloud/server/StatsCollector.java

```diff
@@ -1020,14 +1020,18 @@ protected void runInContext() {
                         if (answer != null && answer.getResult()) {
                             storagePoolStats.put(pool.getId(), (StorageStats)answer);
 
+                            boolean poolNeedsUpdating = false;
                             // Seems like we have dynamically updated the pool size since the prev. size and the current do not match
-                            if (pool.getCapacityBytes() != ((StorageStats)answer).getCapacityBytes() ||
-                                    pool.getUsedBytes() != ((StorageStats)answer).getByteUsed()) {
+                            if (_storagePoolStats.get(poolId) != null && _storagePoolStats.get(poolId).getCapacityBytes() != ((StorageStats)answer).getCapacityBytes()) {
```

Comment:
@rhtyd may be you can update the capacity bytes (from the 'GetStorageStatsCommand' answer) when it is > 0. Otherwise, better to log a WARN message indicating that answer holds the capacity bytes <= 0.

Reply:
The code is reverted to how it was structured in 4.11.3; I suppose at least the solidfire storage pool capacity wouldn't reset to 0. But good idea to add some guarding (either for the 0 check or if the pool is solidfire).